File: differentСontests/2/4.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t;
    cin >> t;
    cin.ignore(); 

    while (t--) {
        int n;
        cin >> n;
        cin.ignore(); 

        vector<int> A(n);
        for (int i = 0; i < n; i++) {
            cin >> A[i];
        }

        unordered_map<int, int> B;
        for (int i = 0; i < n; i++) {
            B[A[i]]++;
        }

        cin.ignore(); 
        string s;
        getline(cin, s);
        
        istringstream iss(s);
        vector<int> tokens;
        string token;
        bool flag = true;

        while (iss >> token) {
            try {
                int x = stoi(token);
                tokens.push_back(x);
            } catch (const invalid_argument &e) {
                flag = false;
                break;
            }
        }

        if (flag && tokens.size() == n) {
            vector<int> sortedTokens = tokens;
            sort(sortedTokens.begin(), sortedTokens.end());
            
            if (tokens != sortedTokens) {
                flag = false;
            }

            for (int num : tokens) {
                if (B.find(num) == B.end()) {
                    flag = false;
                    break;
                }
            }
        } else {
            flag = false;
        }

        if (flag) {
            cout << "YES" << "\n";
        } else {
            cout << "NO" << "\n";
        }
    }

    return 0;
}

File: differentСontests/2/4.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	scanner.Scan()
	t, _ := strconv.Atoi(scanner.Text())

	for t > 0 {
		scanner.Scan()
		n, _ := strconv.Atoi(scanner.Text())

		scanner.Scan()
		line := scanner.Text()
		parts := strings.Fields(line)
		A := make(map[int]int)

		for _, part := range parts {
			num, _ := strconv.Atoi(part)
			A[num]++
		}

		scanner.Scan()
		line = scanner.Text()
		tokens := strings.Fields(line)
		var parsedTokens []int
		flag := true

		for _, token := range tokens {
			x, err := strconv.Atoi(token)
			if err != nil {
				flag = false
				break
			}
			parsedTokens = append(parsedTokens, x)
		}

		if flag && len(parsedTokens) == n {
			sortedTokens := make([]int, len(parsedTokens))
			copy(sortedTokens, parsedTokens)
			sort.Ints(sortedTokens)

			if !equal(parsedTokens, sortedTokens) {
				flag = false
			}

			for _, num := range parsedTokens {
				if _, found := A[num]; !found {
					flag = false
					break
				}
			}
		} else {
			flag = false
		}

		if flag {
			fmt.Println("YES")
		} else {
			fmt.Println("NO")
		}

		t--
	}
}

func equal(a, b []int) bool {
	if len(a) != len(b) {
		return false
	}
	for i := range a {
		if a[i] != b[i] {
			return false
		}
	}
	return true
}
